### PR TITLE
refactor serverless examples documentation

### PR DESCRIPTION
**Pull request by csantanapr**
### What does this PR do?

- Update aws cli commands to output json and specify region
- Adds gitignore files
- Uses envsubt with template files to replace variables
- Add markdown for code commands

### Motivation

Found some problems while running thru the serverless labs

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)

- [ ] Yes, I have added a new example under [examples](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/examples) to support my PR

- [x] Yes, I have updated the [docs](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/doc) for this feature

- [ ] Yes, I have linked to an issue or feature request (applicable to PRs that solves a bug or a feature request)

**Note**:
 - Not all the PRs require examples and docs
 - We prefer small, well tested pull requests. Please ensure your pull requests are self-contained, and commits are squashed

### For Moderators
- [x] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/upbound-aws-provider/composite-resources/serverless-examples/sns-sqs-lambda-s3/README.md [hunk 1]
```diff
@@ -56,22 +57,36 @@ xsqslambdas3.awsblueprints.io                   5m
...
+```
+
+Change the default value for `CLAIM_NAME`
+```shell
+export CLAIM_NAME=test-sns-sqs-lambda-s3
 ```
-sed -i -e "s/replace-with-unique-s3-bucket/$S3_BUCKET/" sns-sqs-lambda-s3-claim.yaml
-sed -i -e "s/replace-with-aws-region/$REGION/" sns-sqs-lambda-s3-claim.yaml
+
+Use the template file `sns-sqs-lambda-s3-claim-tmpl.yaml` to create the claim file `sns-sqs-lambda-s3-claim.yaml` with the variables `CLAIM_NAME`, `S3_BUCKET`, and `REGION` substituted
```
Comment (candonov) [inline examples/upbound-aws-provider/composite-resources/serverless-examples/sns-sqs-lambda-s3/README.md:73]: remove the `sns-sqs-lambda-s3-claim.yaml`

File: examples/upbound-aws-provider/composite-resources/serverless-examples/sqs-lambda-s3/README.md [hunk 2]
```diff
@@ -48,27 +50,42 @@ xsqslambdas3.awsblueprints.io                   5m
...
-sed -i -e "s/replace-with-unique-s3-bucket/$S3_BUCKET/" sqs-lambda-s3-claim.yaml
-sed -i -e "s/replace-with-aws-region/$REGION/" sqs-lambda-s3-claim.yaml
+
+Change the default value for `CLAIM_NAME`
+```shell
+export CLAIM_NAME=test-sqs-lambda-s3
 ```
-Apply the claim
+
+Use the template file `sqs-lambda-s3-claim-tmpl.yaml` to create the claim file `sqs-lambda-s3-claim.yaml` with the variables `CLAIM_NAME`, `S3_BUCKET`, and `REGION` substituted
```
Comment (candonov) [inline examples/upbound-aws-provider/composite-resources/serverless-examples/sqs-lambda-s3/README.md:66]: same here, remove `sqs-lambda-s3-claim.yaml` file

File: examples/upbound-aws-provider/composite-resources/serverless-examples/kinesis-lambda-s3-logs/README.md
Comment (candonov) [inline examples/upbound-aws-provider/composite-resources/serverless-examples/kinesis-lambda-s3-logs/README.md:1]: In the diagram, I think you need to replace SQS with CloudWatch Logs.

Comment (csantanapr) [reply examples/upbound-aws-provider/composite-resources/serverless-examples/kinesis-lambda-s3-logs/README.md:1]: There is no sqs in mermaid
image

Comment (csantanapr) [reply examples/upbound-aws-provider/composite-resources/serverless-examples/kinesis-lambda-s3-logs/README.md:1]: There is no sqs in top block diagram
image

Comment (csantanapr) [reply examples/upbound-aws-provider/composite-resources/serverless-examples/kinesis-lambda-s3-logs/README.md:1]: Oh I see in terms of building blocks

Comment (candonov) [reply examples/upbound-aws-provider/composite-resources/serverless-examples/kinesis-lambda-s3-logs/README.md:1]: Yes, in the image diagram, not the mermaid.

**Comment by csantanapr** [comment]
Address your comments @candonov including the jpg diagram thank you 👍

**Review by candonov — APPROVED** [review]

**Pull request merged**